### DISCUSSION TITLE
Merged Cells not parsed if in "readonly" mode

### DISCUSSION THREAD

**Discussion by MattInventPartners** [PHPOffice/PhpSpreadsheet#2681] [Ideas]
Took me quite a while to find this one.

In Reader/Xlsx.php, we have the following:

Line 877: `if ($xmlSheet && $xmlSheet->mergeCells && $xmlSheet->mergeCells->mergeCell && !$this->readDataOnly) {`

The net effect is that if the sheet is opened in "readonly" mode, it won't find any of the mergedCells, and other methods such as:

`cell::isInMergeRange()`
`cell::isMergeRangeValueCell()`
`cell::getMergeRange() `

...incorrectly return false for merged cells.

**Comment by MarkBaker** [comment]
That's because the "structure" of the spreadsheet isn't data; so it depends on your definition of "incorrectly". `readDataOnly` only loads the data that's contained in the spreadsheet cells, not the "structure" of the spreadsheet, or the styles.
